WooCommerce/Classes/ViewRelated/Coupons/CouponDetails/CouponDetails.swift

```diff
@@ -7,12 +7,21 @@ final class CouponDetailsHostingController: UIHostingController<CouponDetails> {
 
     init(viewModel: CouponDetailsViewModel, onUpdate: @escaping () -> Void, onDeletion: @escaping () -> Void) {
         super.init(rootView: CouponDetails(viewModel: viewModel, onUpdate: onUpdate, onDeletion: onDeletion))
+
         // The navigation title is set here instead of the SwiftUI view's `navigationTitle`
         // to avoid the blinking of the title label when pushed from UIKit view.
         title = viewModel.couponCode
 
         // Set presenting view controller to show the notice presenter here
         rootView.noticePresenter.presentingViewController = self
+
+        // Set manually the edit coupon button click event to present
+        // the AddEditCoupon view on top of the Coupon details
+        rootView.onEditCoupon = { [weak self] addEditCouponViewModel in
+            guard let self = self else { return }
+            let addEditHostingController = AddEditCouponHostingController(viewModel: addEditCouponViewModel)
+            self.navigationController?.present(addEditHostingController, animated: true)
```

Comment:
We don't really need to present the view on top of the navigation controller unless there's a clear reason for that. In this case, I believe we can simply present it from the current controller:
```suggestion
            self.present(addEditHostingController, animated: true)
```

Reply:
Oh, that makes sense! Now it's fixed! https://github.com/woocommerce/woocommerce-ios/pull/6889/commits/610110df70a7cd8dd178d2b8b9fb874b18b20c36